src/openms/include/OpenMS/FILTERING/TRANSFORMERS/SpectraMerger.h

```diff
@@ -651,13 +653,15 @@ namespace OpenMS
       {
         if (merged_indices.count(i) == 0) // save unclustered ones
         {
-          exp_tmp.addSpectrum(exp[i]);
+          exp_tmp.addSpectrum(std::move(exp[i]));
           exp[i] = empty_spec;
         }
       }
 
       //typedef std::vector<typename MapType::SpectrumType> Base;
       //exp.Base::operator=(exp_tmp);
+      //TODO CLEAN THIS UP!
+      //Meta_Data will not be cleared
       exp.clear(false);
       exp.getSpectra().insert(exp.end(), exp_tmp.begin(), exp_tmp.end());
```

Comment:
todo: how to move all spectra, instead of copying

Reply:
https://stackoverflow.com/questions/10720122/is-there-a-standard-way-of-moving-a-range-into-a-vector

Reply:
same in line `exp.getSpectra().insert(exp.end(), merged_spectra.begin(), merged_spectra.end());`